casper/src/main/resources/PoS.rhox

```diff
@@ -366,104 +507,97 @@ in {
+                                        // 5. update `pendingWithdrawers` to empty Map {}.
+                                        removeQuarantinedWithdrawers!(blockNumber, newMovedPendingState, *removeQuarantinedCh) |
+                                        for (@newRemovedQuantinueState <- removeQuarantinedCh) {
+                                          // 6. Pick new `activeValidators`
+                                          pickActiveValidators!(newMovedPendingState.get("allBonds").toList(), *newValidatorsCh) |
+                                          for (@newValidators <- newValidatorsCh) {
+                                            stateUpdateCh!(newRemovedQuantinueState.set("activeValidators", newValidators), (true, Nil))
+                                          }
+                                        }
+                                      }
                                     }
                                   }
-                                } |
+                                } | 
                                 // Transfers withdrawing validator's bond + accumulated rewards from PoS vault to their vault.
-                                new payWithdrawer in {
-                                  contract payWithdrawer(@(pk, amount), returnCh) = {
-                                    new vaultCh, revAddressCh in {
-                                      revAddressOps!("fromPublicKey", pk, *revAddressCh) |
-                                      for (@toRevAddress <- revAddressCh) {
-                                        @posVault!("transfer", toRevAddress, amount, posAuthKey, *returnCh)
-                                      }
+                                contract payWithdrawer(@(pk, amount), returnCh) = {
+                                  new vaultCh, revAddressCh in {
+                                    revAddressOps!("fromPublicKey", pk, *revAddressCh) |
+                                    for (@toRevAddress <- revAddressCh) {
+                                      @posVault!("transfer", toRevAddress, amount, posAuthKey, *returnCh)
+                                    }
+                                  }
+                                } |
+                                contract commitCurrentEpochRewards(@state, @currentRewards, returnCh) = {
+                                  new accumulate in {
+                                    @ListOps!("fold", currentRewards.toList(), state, *accumulate, *returnCh) |
+                                    contract accumulate(@(pk, reward), @updatedState, resultCh) = {
+                                      resultCh!(updatedState.set("comittedRewards",
+                                                                  updatedState.get("comittedRewards").set(pk, updatedState.get("comittedRewards").getOrElse(pk, 0) + reward)))
+                                    }
```

Comment:
Should we just remove all committed reward map?

Reply:
We could remove committed reward map if we return rewards to validators at epoch instead of keeping it which I think it is more economic friendly.

Reply:
Or, we can accumulate the rewards into bonds which is more like coop's current staking plan.

Reply:
I was thinking about the place where committed rewards can be cleared.

Let's keep existing logic and accumulate rewards until the epoch when validator will withdraw his stake + rewards.